src/main/kotlin/com/openlattice/mechanic/upgrades/V3TimeUseDiaryUpgrade.kt

```diff
@@ -0,0 +1,437 @@
+                }
+                if (studyId == null && answer.studyId != null) {
+                    studyId = answer.studyId
+                }
+                if (participant == null && answer.participant != null) {
+                    participant = answer.participant
+                }
+                if (submissionDate == null && answer.submissionDate != null) {
+                    submissionDate = answer.submissionDate
+                }
+                if (organizationId == null && answer.organizationId != null) {
+                    organizationId = answer.organizationId
+                }
+            }
+
+            if (submissionId == null || studyId == null || participant == null || submissionDate == null || organizationId == null) {
+                logger.warn("Skipping Submission for answer $answers. " +
+                        "Failed to retrieve required detail for Submission from answer group: submissionId, submissionDate, studyId, or participantId.")
+                continue
+            }
+            submissions.add(
+                Submission(
+                    submissionId,
+                    organizationId,
+                    studyId,
+                    participant,
+                    submissionDate,
+                    submissionResponses
+                )
+            )
+        }
+        return submissions
+    }
+
+    private fun getChronicleSuperUserPrincipals(): Set<Principal> {
+        return chronicleSuperUserIds
+            .map { principalService.getSecurablePrincipal(it)}
+            .map { principalService.getAllPrincipals(it).map { principal -> principal.principal } }
+            .flatten().toSet()
+    }
+
+    private fun fstStr(collection: Collection<Any>?): String? {
+        collection ?: return null
+        return collection.first().toString()
+    }
+
+    override fun getSupportedVersion(): Long {
+        return Version.V2021_07_23.value
+    }
+}
+
+data class AnswerNeighborhood(
+    val submissionId:       UUID?,
+    var organizationId:     UUID?,
+    var studyId:            UUID?,
+    val participant:        String?,
+    val submissionDate:     OffsetDateTime?,
+    val questionString:     String?,
+    val questionCode:       String?,
+    var answer:             Set<String>?,
+    val startDateTime:      OffsetDateTime?,
+    val endDateTime:        OffsetDateTime?,
+    val participantId:      UUID?,
+    val participantESID:    UUID?,
+)
+
+data class TimeUseDiaryResponse(
+    val code: String,
+    val question: String,
+    val response: Set<String>,
+    val startDateTime: OffsetDateTime?,
+    val endDateTime: OffsetDateTime?,
+)
+
+data class Submission(
+    val submissionId: UUID,
+    val organizationId: UUID,
+    val studyId: UUID,
+    val participant: String,
+    val submissionDate: OffsetDateTime,
+    val submission: List<TimeUseDiaryResponse>
+)
+
+class OffsetDateTimeSerializer<OffsetDateTime>() : JsonSerializer<OffsetDateTime>() {
```

Comment:
Why does this matter? Unless we're storing a timezone field once it's in postgres timezone information gets lost.

Reply:
This is added as serializer module on the objectMapper to prevent datetime fields in TimeUseDiaryReponses from being converted to epoch seconds + nanos when using objectMapper.writeValueAsString to set the JSONB preparedstatement field.

Although upon double checking, I see that the same result is achieved more simply by adding `.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)` to the objectMapper.